caravel/assets/javascripts/SqlLab/components/CopyQueryTabUrl.jsx

```diff
@@ -33,16 +28,21 @@ export default class CopyQueryTabUrl extends React.Component {
     if (qe.sql) params.push('sql=' + encodeURIComponent(qe.sql));
 
     const queryString = params.join('&');
-    const queryLink = this.state.cleanUri + '?' + queryString;
+    const queryLink = window.location.pathname + '?' + queryString;
+    getShortUrl(queryLink, this.onShortUrlSuccess.bind(this));
+  }
 
-    return queryLink;
+  onShortUrlSuccess(data) {
```

Comment:
i'm a big fan of more, smaller methods, rather than putting everything in line. this seems cleaner than putting in directly in `componentWillMount`, imo.